Scripts/Mac/BuildInstaller.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+CONFIGURATION=$1
+if [ -z $CONFIGURATION ]; then
+  CONFIGURATION=Debug
+fi
+
+PACKAGEVERSION=$2
+if [ -z $PACKAGEVERSION ]; then
+  PACKAGEVERSION="1.0.0.1"
+fi
+
+SCRIPTDIR="$(dirname ${BASH_SOURCE[0]})"
+
+# convert to an absolute path because it is required by `dotnet publish`
+pushd $SCRIPTDIR
+SCRIPTDIR="$(pwd)"
+popd
+
+SRCDIR=$SCRIPTDIR/../..
+ROOTDIR=$SRCDIR/..
+BUILDOUTPUT=$ROOTDIR/BuildOutput
+PUBLISHDIR=$ROOTDIR/Publish
+INSTALLROOT=$ROOTDIR/InstallRoot
+VFSFORGITDESTINATION="usr/local/vfsforgit"
+INSTALLERPACKAGENAME="VFSForGit.$PACKAGEVERSION"
+
+echo "SRCDIR: ${SRCDIR}"
+echo "ROOTDIR: ${ROOTDIR}"
+echo "BUILDOUTPUT: ${BUILDOUTPUT}"
+echo "PUBLISHDIR: ${PUBLISHDIR}"
+
+function CheckBuildIsAvailable()
+{
+	if [ ! -d $BUILDOUTPUT ] || [ ! -d $PUBLISHDIR ]; then
+  		echo "Could not find VFSForGit Build to package."
+  		exit 1
+	fi
+}
+
+function SetPermissions()
+{
+	chmodCommand="chmod -R 755 \"${INSTALLROOT}\""
+	echo $chmodCommand
+	eval $chmodCommand || exit 1
+}
+
+function CreateInstallerRoot()
+{
+	mkdirVfsForGit="mkdir -p \"${INSTALLROOT}/$VFSFORGITDESTINATION\""
+	echo $mkdirVfsForGit
+	eval $mkdirVfsForGit || exit 1
+	
+	mkdirBin="mkdir -p \"${INSTALLROOT}/usr/local/bin\""
+	echo $mkdirBin
+	eval $mkdirBin || exit 1
+}
+
+function CopyBinariesToInstall()
+{
+	copyPublishDirectory="cp -Rf \"${PUBLISHDIR}\"/* \"${INSTALLROOT}/${VFSFORGITDESTINATION}/.\""
+	echo $copyPublishDirectory
+	eval $copyPublishDirectory || exit 1
+	
+	removeTestAssemblies="find \"${INSTALLROOT}/${VFSFORGITDESTINATION}\" -name \"*GVFS.*Tests*\" -exec rm -f \"{}\" \";\""
+	echo $removeTestAssemblies
+	eval $removeTestAssemblies || exit 1
+	
+	removeDataDirectory="rm -Rf \"${INSTALLROOT}/${VFSFORGITDESTINATION}/Data\""
+	echo $removeDataDirectory
+	eval $removeDataDirectory || exit 1
+	
+	copyKext="cp -Rf \"${BUILDOUTPUT}/ProjFS.Mac/Native/Build/Products/$CONFIGURATION\"/* \"${INSTALLROOT}/${VFSFORGITDESTINATION}/.\""
+	echo $copyKext
+	eval $copyKext || exit 1
+	
+	currentDirectory=`pwd`
+	cd "${INSTALLROOT}/usr/local/bin"
+	linkCommand="ln -s ../vfsforgit/gvfs gvfs"
+	echo $linkCommand
+	eval $linkCommand
+	cd $currentDirectory
+}
+
+function CreateInstaller()
+{
+	pkgBuildCommand="/usr/bin/pkgbuild --identifier com.microsoft.vfsforgit.pkg --root \"${INSTALLROOT}\" \"${ROOTDIR}\"/$INSTALLERPACKAGENAME.pkg"
+	echo $pkgBuildCommand
+	eval $pkgBuildCommand || exit 1
+}
+
+function CreateDiskImage
```

Comment:
I'm against the pattern of creating a .dmg file to house a .pkg. It's a relic of the old days when .pkg files used to not be flat files (they used to be packages/bundles, see https://en.wikipedia.org/wiki/Installer_(macOS)#Installer_package). These days, all we get is a very minor compression in return for increasing the amount of clicks needed for a user to install our software. @pmj might also have opinions on MacOS installers that are different than my experiences, so I'm curious to know where he stands on this.

https://apple.stackexchange.com/questions/30695/why-are-so-many-mac-downloads-a-dmg-just-that-contains-a-single-pkg

FWIW: I think this should also be changed in our Git installer, but that's another story :)

Reply:
Unless there's a good reason to do otherwise, my vote is for a standalone .pkg as well.

Reply:
Removed DMG.